### PR TITLE
refactor: Unconditionally use blocked mult in EigenStepper

### PR DESCRIPTION
**Pull request by paulgessinger**
Just to try if this also returns clean outputs, I'm downgrading the `if` added in #1181 to an `assert`. Let's see what the CI says here.

### CODE REVIEW

**Comment by stephenswat** [comment]
Nice!

@andiwand do you have any performance numbers for this?

**Comment by paulgessinger** [comment]
@stephenswat I can run a quick test.

**Comment by paulgessinger** [comment]
This is from 10 runs:

image

File: Core/include/Acts/Propagator/EigenStepper.ipp [hunk 1]
```diff
@@ -250,24 +250,20 @@ Acts::Result<double> Acts::EigenStepper<E, A>::step(
...
-      // For safety purposes, we provide a full matrix multiplication as a
-      // backup strategy.
-      state.stepping.jacTransport = D * state.stepping.jacTransport;
-    }
+    assert((D.topLeftCorner<4, 4>().isIdentity()) &&
+           (D.bottomLeftCorner<4, 4>().isZero()) &&
+           (state.stepping.jacTransport.template topLeftCorner<4, 4>()
+                .isIdentity()) &&
+           (state.stepping.jacTransport.template bottomLeftCorner<4, 4>()
+                .isZero()));
```
Comment (andiwand) [inline Core/include/Acts/Propagator/EigenStepper.ipp:258]: nitpick: should we make multiple asserts for this?

**Review by andiwand — APPROVED** [review]
very interesting! I thought the assert would fire

File: Core/include/Acts/Propagator/EigenStepper.ipp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Propagator/EigenStepper.ipp:258]: Can do

**Comment by paulgessinger** [comment]
image

More stats confirm the measurement from before, I think we're looking at about a 5% speedup.

**Comment by paulgessinger** [comment]
Athena is green now, should we get this in? @andiwand

**Review by andiwand — APPROVED** [review]
lets go

**Pull request merged**